src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/disable_gathercompressed_quantized_model.cpp

```diff
@@ -0,0 +1,130 @@
+    }
+
+protected:
+    void SetUp() override {
+        targetDevice = ov::test::utils::DEVICE_CPU;
+        element::Type weight_prec;
+        InputShape inputShape1, inputShape2;
+        std::tie(weight_prec, inputShape1, inputShape2) = GetParam();
+
+        // auto input_shape1 = Shape{1, 3, 64, 64};
+        // auto input_shape2 = Shape{32};
+        init_input_shapes({inputShape1, inputShape2});
+
+        targetDevice = test::utils::DEVICE_CPU;
+        auto type = element::f32;
+
+        auto input1 = std::make_shared<op::v0::Parameter>(type, inputDynamicShapes[0]);
+        auto input2 = std::make_shared<op::v0::Parameter>(type, inputDynamicShapes[1]);
+
+        auto shared_il = op::v0::Constant::create(type, {1, 1, 1, 1}, {0.f});
+        auto shared_ih = op::v0::Constant::create(type, {1, 1, 1, 1}, {12.5f});
+        auto shared_ol = op::v0::Constant::create(type, {1, 1, 1, 1}, {0.f});
+        auto shared_oh = op::v0::Constant::create(type, {1, 1, 1, 1}, {12.5f});
+        auto fq = std::make_shared<op::v0::FakeQuantize>(input1, shared_il, shared_ih, shared_ol, shared_oh, 256);
+
+        // Weights
+        auto weights_shape = Shape{64, 64};
+        auto weights_vals = test::utils::generate_float_numbers(shape_size(weights_shape), -1, 1);
+        auto weights = op::v0::Constant::create(weight_prec, weights_shape, weights_vals);
+        auto convert = std::make_shared<op::v0::Convert>(weights, element::f32);
+        auto multiply = std::make_shared<op::v1::Multiply>(convert, op::v0::Constant::create(type, {1, 1}, {0.625}));
+        // Indics
+        auto softmax = std::make_shared<op::v1::Softmax>(input2, 0);
+        auto multiply2 = std::make_shared<op::v1::Multiply>(softmax, op::v0::Constant::create(type, {1}, {64}));
+        auto indics = std::make_shared<op::v0::Convert>(multiply2, element::i64);
+        // Gather
+        auto gather =
+            std::make_shared<op::v8::Gather>(multiply, indics, op::v0::Constant::create(element::i32, Shape{1}, {0}));
+
+        auto matMul = std::make_shared<ov::op::v0::MatMul>(fq, gather, false, true);
+
+        function = std::make_shared<Model>(matMul, ParameterVector{input1, input2});
+    }
+
+    void check_results() {
+        const auto& test_param = GetParam();
+        const auto compressed_weights_precision = std::get<0>(test_param);
+
+        const auto runtime_model = compiledModel.get_runtime_model();
+        const auto result = runtime_model->get_result();
+        const auto matmul = result->get_input_node_shared_ptr(0);
+        if (compressed_weights_precision == element::i8) {
+            EXPECT_EQ(matmul->get_input_element_type(1), element::i8);
+        } else if (compressed_weights_precision == element::u8) {
+            // oneDNN MutMul support precision: Source(u8, s8), Weights(s8)
+            // So reorder will be inserted when weights is not s8.
+            const auto mm_productor = matmul->get_input_node_shared_ptr(1);
```

Comment:
Check in 3 cases:
1: We expected case;
2: Because oneDNN MutMul support precision: Source(u8, s8), Weights(s8), so reorder will be inserted, so I also need to check reorder and expected precision.
3: About precison i4/u4, I don't know if we need to do. Because they have no changes vs master branch.
@dmitry-gorokhov

Reply:
I think it is not necessary to check precision of Reorder input. Instead you should check runtime precision (exec_model_info::RUNTIME_PRECISION) of Matmul (which is expected to be I8).

Reply:
When `compressed_weights_precision == element::u8,` reorder will be inserted, the output precision of reorder is not `I8`, it is `f32`, so the `MatMul's exec_model_info::RUNTIME_PRECISION` also is` f32`.
So I will update code to check MatMul's input(1) precision is `f32`.
@dmitry-gorokhov @v-Golubev

Reply:
What you are describing means Matmul is executed in fp32 precision and not in Int8 which is not exepcted. We need to figure out why it happening

Reply:
oneDNN MutMul official support precisions don't contain (u8+u8).
https://oneapi-src.github.io/oneDNN/dev_guide_matmul.html#data-types

Here is debug log about `getSupportedPrimitiveDescriptors`
```
[ DEBUG ] graph.cpp:400 InitDescriptors() #15 MatMul_649_original  SupportedPrimitiveDescriptors [0/1]: 
  brgemm_avx512(,{0 - ?, 3, 0 - ?, 0 - ?} f32 abcd inPlace:-1 non-constant,{1, 1, 32, 64} f32 abcd inPlace:-1 non-constant) -> (,{0 - ?, 3, 0 - ?, 32} f32 abcd inPlace:-1 non-constant)
```
So 2 Reorders(u8->f32) are inserted.

Reply:
@xipingyan My mistake, it is indeed limitation on oneDNN side. 
Could you please make Emdedding precision equal to I8 (instead of I8). It should solve the problem

Reply:
Embedding precision with 4 types(I8,U8,I4,U4) is tested. You mean we just only test embedding precision I8?
I think we can keep current test, and revert initial version(just check reorder precision is U8 when embedding precision is I8). @dmitry-gorokhov What's your idea?

Reply:
There should be different conditions:
1. For I4/U4 lets check presence of GatherCompressed operation n  the runtime graph 
2. For I8/U8 lets check presence of Gather operation in the runtime graph 
3. Additionaly for I8 lets check that Matmul has I8 runtime precision
4. U8 case lets leave w/o additional checks

Reply:
Updated,
BTW, for I8 weights, MatMul's RUNTIME_PRECISION is U8, because FakeQuantize(matmul's input(0))'s output precision is u8.